Assets/MixedRealityToolkit-SDK/Features/SpatialAwareness/System/MixedRealitySpatialAwarenessSystem.cs

```diff
@@ -0,0 +1,572 @@
+
+        private float updateInterval = 3.5f;
+
+        /// <inheritdoc />
+        public float UpdateInterval
+        {
+            get
+            {
+                return updateInterval;
+            }
+
+            set
+            {
+                if (IsObserverRunning)
+                {
+                    throw new System.InvalidOperationException("UpdateInterval cannot be modified while the observer is running.");
+                }
+
+                updateInterval = value;
+            }
+        }
+
+        /// <inheritdoc />
+        public bool IsObserverRunning { get; private set; } = false;
+
+        /// <inheritdoc />
+        public void ResumeObserver()
+        {
+            if (IsObserverRunning) { return; }
+
+#if UNITY_WSA
+            // todo
+#endif // UNITY_WSA
+
+            IsObserverRunning = true;
+        }
+
+        /// <inheritdoc />
+        public void SuspendObserver()
+        {
+            if (!IsObserverRunning) { return; }
+
+#if UNITY_WSA
+            // todo
+#endif // UNITY_WSA
+
+            IsObserverRunning = false;
+        }
+
+#region Mesh Handling implementation
+
+        /// <inheritdoc />
+        public bool UseMeshSystem { get; set; } = true;
+
+        /// <inheritdoc />
+        public int MeshPhysicsLayer { get; set; } = 31;
+
+        /// <inheritdoc />
+        public int MeshPhysicsLayerMask => 1 << MeshPhysicsLayer;
+
+        private SpatialAwarenessMeshLevelOfDetail meshLevelOfDetail = SpatialAwarenessMeshLevelOfDetail.Coarse;
+
+        /// <inheritdoc />
+        public SpatialAwarenessMeshLevelOfDetail MeshLevelOfDetail
+        {
+            get
+            { 
+                return meshLevelOfDetail;
+            }
+
+            set
+            {
+                if (IsObserverRunning)
+                {
+                    throw new System.InvalidOperationException("MeshLevelOfDetail cannot be modified while the observer is running.");
+                }
+
+                if (meshLevelOfDetail != value)
+                {
+                    // Non-custom values automatically modify MeshTrianglesPerCubicMeter
+                    if (value != SpatialAwarenessMeshLevelOfDetail.Custom)
+                    {
+                        meshTrianglesPerCubicMeter = (int)value;
+                    }
+
+                    meshLevelOfDetail = value;
+                }
+            }
+        }
+
+        private int meshTrianglesPerCubicMeter = (int)SpatialAwarenessMeshLevelOfDetail.Coarse;
+
+        /// <inheritdoc />
+        public int MeshTrianglesPerCubicMeter
+        {
+            get
+            {
+                return meshTrianglesPerCubicMeter;
+            }
+
+            set
+            {
+                if (IsObserverRunning)
+                {
+                    throw new System.InvalidOperationException("MeshTrianglesPerCubicMeter cannot be modified while the observer is running.");
```

Comment:
Let's try to keep throwing exceptions to a minimun as they crash the player/editor. Let's raise an error instead.